Realm/RLMObject.h

```diff
@@ -278,6 +278,21 @@ RLM_ASSUME_NONNULL_BEGIN
  */
 + (NSArray RLM_GENERIC(NSString *) *)requiredProperties;
 
+/**
+ Implement to return a dictionary providing information related to linking objects properties.
+
+ Properties of type RLMLinkingObjects must have a corresponding entry in the dictionary to provide
+ information about the origin of the link that they represent. Their corresponding value in the
+ dictionary must be of the form:
+
+     @{ @"class": @"SomeClass", @"property": @"someProperty" }
+
+ Where `SomeClass` is a class that has a property `someProperty` that forms a relationship with this class.
+
```

Comment:
Dictionaries with a fixed set of keys are somewhat of an API antipattern. `[RLMLinkDescriptor descriptorWithClass:MyClass.class property:@"someProperty"]` (or `[RLMLinkDescriptor descriptorWithClassName:@"MyClass" property:@"someProperty"]`) would be a bit less error-prone.

An array of strings using keypath syntax would be a lot less verbose, but probably not a gain overall.